### PR TITLE
Convert path to URL before import of config/plugin for Windows support

### PR DESCRIPTION
**Pull request by lifeart**
fixes https://github.com/ember-template-lint/ember-template-lint/issues/2336

I'm not sure how to test it, because chain for this function is quite long, should we create unit test for 'requirePlugin' function itself?

### CODE REVIEW

File: lib/get-config.js [hunk 1]
```diff
@@ -31,7 +32,7 @@ async function requirePlugin(workingDir, pluginName, fromConfigPath) {
   let pluginPath = resolve.sync(pluginName, { basedir, extensions: ['.js', '.mjs', '.cjs'] });
 
   try {
-    const { default: plugin } = await import(pluginPath);
+    const { default: plugin } = await import(pathToFileURL(pluginPath));
```
Comment (bmish) [inline lib/get-config.js:35]: Wouldn't we also need this for the config loading on line 84 too?

Comment (lifeart) [reply lib/get-config.js:35]: good catch! pushed

File: test/acceptance/cli-test.js [hunk 2]
```diff
@@ -442,6 +442,37 @@ describe('ember-template-lint executable', function () {
     });
   });
 
+  describe('multiplatform config resolve', function () {
+    describe('given absolute config path and path to single file without errors', function () {
```
Comment (bmish) [inline test/acceptance/cli-test.js:446]: Question, is this different from the existing tests for `--config-path`? Does it actually test a Windows-style path like `C:\Users\...`. And does it fail before the fix?

Comment (lifeart) [reply test/acceptance/cli-test.js:446]: it should test windows-style path in our windows CI env
no failures before

Comment (lifeart) [reply test/acceptance/cli-test.js:446]: also, it should fail here: https://github.com/ember-template-lint/ember-template-lint/blob/master/test/helpers/fake-project.js#L68
but not failing, we could experiment with some prefixes ('/' before 'C'), or node versions to catch this error

Comment (lifeart) [reply test/acceptance/cli-test.js:446]: in original bugreport #2336 it's failing for custom plugins (as I understand), likely we could also try to test custom plugin here..

File: lib/get-config.js [hunk 3]
```diff
@@ -31,7 +32,7 @@ async function requirePlugin(workingDir, pluginName, fromConfigPath) {
   let pluginPath = resolve.sync(pluginName, { basedir, extensions: ['.js', '.mjs', '.cjs'] });
```
Comment (bmish) [inline lib/get-config.js:32]: Could the reason we end up needing this be related to `resolve.sync` usage here? I wonder if we can solve this by removing the `resolve.sync` somehow instead? 

Although this link shows your similar workaround: https://github.com/nodejs/help/issues/3380

Also, I'm still hoping we can figure out actual tests for this that fail beforehand.

**Comment by charlesfries** [comment]
Anything I can do to help get this merged

File: lib/get-config.js (hunk 3 above)
Comment (lifeart) [reply lib/get-config.js:32]: @bmish you mean change this logic in scope of this PR? (it's already in codebase)

**Comment by locks** [comment]
@bmish @lifeart what's the latest status on this?

**Comment by bmish** [comment]
From what I remember, we had trouble reproducing the issue in the test and so I wasn't confident that the test actually verified it was fixed.

**Comment by lflo5727** [comment]
Is there any update on this PR? 
This issue is blocking us from adding a custom ember-template-lint plugin. We have the same issue described here: https://github.com/ember-template-lint/ember-template-lint/issues/2336 

If not is there any decent workaround?

**Comment by bmish** [comment]
If someone can confirm that this PR fixes the issue, we might be able to move forward with it.

**Comment by lflo5727** [comment]
@bmish I just pulled this branch down for ember-template-lint and tested. This absolutely fixes the issue. 
Would be great to get this pushed along

**Review by bmish — APPROVED** [review]
Thanks! Let's give this a try.

**Pull request merged**